bin/cylc-show-times

```diff
@@ -0,0 +1,372 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""cylc [util] show-times [OPTIONS] REG
+
+Retrieve timing information for the given cylc suite that can be
+used for fine-grain diagnostics for wait time and run time performance
+analysis.  Two main categories of output are available: summary output
+(in text or web format) and raw output.  If an output filename is
+supplied, output is redirected there otherwise both summary and raw
+outputs are printed to standard output.
+
+Summary Output (the default):
+Data stratified by host and batch system that provides a statistical
+summary of
+    1. Queue wait time (duration between task submission and start times)
+    2. Task run time (duration between start and succeed times)
+    3. Total run time (duration between task submission and succeed times)
+Text output will show summary tables in text only.  Web output generates
+HTML that contains boxplots as well as summary tables suitable for viewing
+in a browser.  Both summary options require the use of the Pandas library,
+and the web summary output requires the Matplotlib plotting library.  If
+these libraries are not available, an error message will be printed and the
+summary will not be generated.
+
+Raw Output:
+A flat list of tabular data that provides (for each task and cycle) the
+    1. Time of successful submission
+    2. Time of task start
+    3. Time of task successful completion
+as well as information about the batch system and remote host to permit
+stratification/grouping if desired by downstream processors.
+
+Timings are shown only for succeeded tasks.
+
+For long-running and/or large suites (i.e. for suites with many task events),
+the database query to obtain the timing information may take some time.
+
+"""
+
+import sys
+from cylc.remote import remrun
+if remrun().execute():
+    sys.exit(0)
+
+import cStringIO as StringIO
+import contextlib
+import os
+
+import cylc.flags
+from cylc.cfgspec.globalcfg import GLOBAL_CFG
+from cylc.option_parsers import CylcOptionParser as COP
+from cylc.rundb import CylcSuiteDAO
+
+
+@contextlib.contextmanager
+def smart_open(filename=None):
+    """
+    Allow context management for output to either a file or
+    to standard output transparently.
+
+    See https://stackoverflow.com/a/17603000
+
+
+    """
+    if filename and filename != '-':
+        fh = open(filename, 'w')
+    else:
+        fh = sys.stdout
+    try:
+        yield fh
+    finally:
+        if fh is not sys.stdout:
+            fh.close()
+
+
+def main():
+    parser = COP(
+        __doc__,
+        argdoc=[('REG', 'Suite name')]
+    )
+    parser.add_option(
+        "-r", "--raw",
+        help="Show raw timing output for downstream processing.",
```

Comment:
"downstream processing" meaning user-specific postprocessing that happens with the data.  This is less necessary now that we're including some sort of summary capability within the command, but I can still see people wanting to do something extra and using the raw output as data input.